File: file-lock/lock.c
#include <sys/file.h>
#include <stdio.h>
#include <unistd.h>

int main (void)
{
    FILE *fp = NULL;
    int i = 20;
    if ((fp = fopen("./l", "r+b")) == NULL)
        printf("file open error!\n");

    printf("wait lock\n");
    if (flock(fp->_fileno, LOCK_EX) != 0)
        printf("file lock by others\n");

    printf("get lock\n");
    sleep(20);
    fclose(fp);
    flock(fp->_fileno, LOCK_UN);
    printf("unlock\n");
    return 0;
}

File: file-lock/lock.py
#coding:utf-8

import fcntl
import time

fp = open('./l','r')
print('wait to lock')
fcntl.flock(fp, fcntl.LOCK_EX)
print('get lock')
time.sleep(20)
fcntl.flock(fp, fcntl.LOCK_UN)
print('unlock')
fp.close()
